src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -360,6 +365,8 @@ protected HdfsSinkConnectorConfig(ConfigDef configDef, Map<String, String> props
           e
       );
     }
+
+    partsToReplace = getPartsToReplace();
```

Comment:
Why does `partsToReplace` need to be a field?

Reply:
micro optimization to not compile the regex pattern and perform the regex search every call, but I just got a better idea - Ill just keep track of the largest index instead